src/compiler/emitter.ts

```diff
@@ -4156,9 +4161,13 @@ namespace ts {
         }
 
         function emitModifiers(node: Node, modifiers: NodeArray<Modifier> | undefined) {
-            if (modifiers && modifiers.length) {
-                emitList(node, modifiers, ListFormat.Modifiers);
-                writeSpace();
+            if (modifiers) {
+                onBeforeEmitNodeArray?.(modifiers);

```

Comment:
I was getting an error previously because `onBeforeEmitNodeArray` (/`onAfter...`) was not called for an empty modifiers array. The problem is that the text changes printer uses those two handlers to set positions in synthetic nodes, when we are emitting, so empty modifier arrays were not having their positions set. In `emitNodeList`, we call those handlers even if the node array is empty, so it seemed to make sense to have the same behavior here, for consistency, instead of coming up with some alternative fix.

Reply:
Possibly dumb question, but when/why do we have empty modifier arrays?

Reply:
It's a bit of a messy situation: in this completion scenario, I was getting empty modifier array because I call `updateModifiers`, which in turns calls `createModifiersFromModifierFlags`, which always returns an array, even if empty.

Reply:
@rbuckton should `updateModifiers` pass through `undefined` instead of an empty array to represent no modifiers?

Reply:
I think we generally expect modifiers to be either `undefined` or non-empty, though we mostly check for `node.modifiers?.length` (or `node.modifiers && node.modifiers.length`). There's no reason to have an empty array here, so it was probably an oversight when I added `createModifiersFromModifierFlags`. We should probably always put an `undefined` for empty modifiers, but always defensively check for `node.modifiers?.length` (or use `some(node.modifiers)`, which essentially does the same thing)

Reply:
updated this to revert the emitter change to `emitModifiers` and to change `createModifiersFromModifierFlags` to return `undefined` instead of an empty array. this changed the API, since now `createModifiersFromModifierFlags` returns `Modifiers[] | undefined`.